### PR TITLE
Create home page functionality

### PR DESCRIPTION
**Pull request by MReyna12**

### CODE REVIEW

**Comment by MReyna12** [comment]
I believe I have a fix for the types required for this function:
```typescript
const createFetchResponse = (data) => {
  return {
    ok: true,
    json: async () => await Promise.resolve(data)
  }
}
```
However, before I implement that I would like to discuss said fix, as well as understanding some of the behavior that occurs when using the debugger and testing. I will add this to our parking lot on Monday.

**Review by rnathuji — APPROVED** [review]
I think this looks good / seems to behave as expected. I had one minor comment to potentially consider.

File: frontend/src/pages/index/+Page.tsx [hunk 1]
```diff
@@ -1,8 +1,334 @@
...
+
+  const handleCourseBuildFormSubmit = async (values: CourseBuildFormValues): Promise<void> => {
+    try {
+      await ropeApi.createCourseBuild(
+        values.instructorFirstName,
+        values.instructorLastName,
+        values.instructorEmail,
+        values.schoolDistrictName
+      )
+      setCourseBuildFormMessage('Course build successfully created!')
```
[the remark this answers is not in the document]
Comment (MReyna12) [reply frontend/src/pages/index/+Page.tsx:197]: We could do that. Would we then want the user to be able to submit a new build for a different instructor without refreshing the page? Aka re-enable the form when the user presses the submit button when searching a moodle instructor. Without building any additional functionality / checks, you could still end up with trying to submit an already built course if you don't type a new moodle user email before hitting the submit button.

[the remark this answers is not in the document]
Comment (MReyna12) [reply frontend/src/pages/index/+Page.tsx:197]: Actually, ignore the above. I can disable the form and we shouldn't run into the problem I noted above.

[the remark this answers is not in the document]
Comment (rnathuji) [reply frontend/src/pages/index/+Page.tsx:199]: :+1: -- this looks good to merge to me.

**Pull request merged**